pyqtgraph/graphicsItems/InfiniteLine.py

```diff
@@ -240,3 +362,122 @@ def setMouseHover(self, hover):
         else:
             self.currentPen = self.pen
         self.update()
+
+    def update(self):
+        # new in version 0.9.11
+        UIGraphicsItem.update(self)
+        br = UIGraphicsItem.boundingRect(self) # directly in viewBox coordinates
+        xmin, ymin, xmax, ymax = br.getCoords()
+        if self.angle == 90:  # vertical line
+            diffX = xmax-xmin
+            diffMin = self.value()-xmin
+            limInf = self.shift*diffX
+            ypos = ymin+self.location*(ymax-ymin)
+            if diffMin < limInf:
+                self.text.anchor = Point(self.anchorRight)
+            else:
+                self.text.anchor = Point(self.anchorLeft)
+            fmt = " x = " + self.format
+            if self.unit is not None:
+                fmt = fmt + self.unit
+            self.text.setText(fmt.format(self.value()), color=self.textColor)
+            self.text.setPos(self.value(), ypos)
+        elif self.angle == 0:  # horizontal line
+            diffY = ymax-ymin
+            diffMin = self.value()-ymin
+            limInf = self.shift*(ymax-ymin)
+            xpos = xmin+self.location*(xmax-xmin)
+            if diffMin < limInf:
+                self.text.anchor = Point(self.anchorUp)
+            else:
+                self.text.anchor = Point(self.anchorDown)
+            fmt = " y = " + self.format
+            if self.unit is not None:
+                fmt = fmt + self.unit
+            self.text.setText(fmt.format(self.value()), color=self.textColor)
+            self.text.setPos(xpos, self.value())
+
+
+    def showLabel(self, state):
+        """
+        Display or not the label indicating the location of the line in data
+        coordinates.
+
+        ==============   ==============================================
+        **Arguments:**
+        state            If True, the label is shown. Otherwise, it is hidden.
+        ==============   ==============================================
+        """
+        if state:
+            self.text.show()
+        else:
+            self.text.hide()
+        self.update()
+
+    def setLocation(self, loc):
```

Comment:
`setLocation` and `setShift` are ambiguous names (especially compared to `setPos`). These should make it clear that the *label* will be affected, like `setLabelLocation` or `setTextLocation`.  For further simplicity, it would be nice for these two methods to be combined into one, like `setTextLocation([0.8, 1.0])`, where the first value is always along the line axis, and the second is always orthogonal (or however you think is most intuitive).